mat/vector.go

```diff
@@ -97,11 +97,11 @@ func NewVecDense(n int, data []float64) *VecDense {
 	}
 }
 
-// SliceVec returns a new VecDense that shares backing data with the receiver.
+// SliceVec returns a new Vector that shares backing data with the receiver.
 // The returned matrix starts at i of the receiver and extends k-i elements.
 // SliceVec panics with ErrIndexOutOfRange if the slice is outside the capacity
 // of the receiver.
-func (v *VecDense) SliceVec(i, k int) *VecDense {
+func (v *VecDense) SliceVec(i, k int) Vector {
```

Comment:
Why this change?

Reply:
Interface satisifaction; if there are multiple `SliceVec` implementers they can only satisfy an interface if there is an equal signature. This is what we did with `Slice` on `Dense`. Maybe it needs rediscussion?